cmd/ebrelayer/txs/parser.go

```diff
@@ -42,8 +39,14 @@ func ParsePayload(valAddr sdk.ValAddress, event *events.LockEvent) (ethbridgeTyp
 		return witnessClaim, errors.New("empty recipient address")
 	}
 
+	// Symbol formatted to lowercase
+	symbol := strings.ToLower(event.Symbol)
+	if symbol == "eth" && event.Token != common.HexToAddress("0x0000000000000000000000000000000000000000") {
```

Comment:
do we want this kind of check in the solidity code too?

Reply:
the corresponding check in the solidity code is on lines 110-114:
```if (msg.value != 0) {
          require(_token == address(0));
          require(msg.value == _amount);
          symbol = "ETH";
        } else { ...
```
if the symbol is 'ETH', then the given token address must be the null address. this check in parser.go after the fact is just to confirm the parameters match up with the contract's restrictions around field `symbol`.

Reply:
The check only happens inside the  if (msg.value != 0) block. Could I not create a new ERC20 token which has the symbol ETH, and so avoid that check? I suppose this would still work and be secure, but unexpected/weird and we should probably block tokens which have the symbol 'ETH' to avoid it.